### DISCUSSION TITLE
ES6 module support

### DISCUSSION THREAD

**Discussion by ithilbor** [prisma/prisma#12935] [Q&A] [answered]
### Bug description

I've followed the instructions on your official docs (that i think are also well written) but when it comes to import the PrismaClient with the "type": "module" in the package.json it doesn't work at all. The wrong instructions are:

```
import { PrismaClient } from '@prisma/client'

const prisma = new PrismaClient()
```

This is my package.json:
```
{
  "name": "test",
  "version": "1.0.0",
  "description": "test",
  "type": "module",
  "main": "server.ts",
  "repository": "unknown",
  "author": "unknown",
  "license": "GPL-3.0-or-later",
  "private": true,
  "dependencies": {
    "@prisma/client": "3.12.0",
    "cors": "^2.8.5",
    "dotenv": "^16.0.0",
    "express": "^4.17.3"
  },
  "scripts": {
    "build": "yarn tsc",
    "dev": "concurrently \"yarn tsc --watch\" \"nodemon -q dist/server.js\"",
    "start": "node ./dist/server.js"
  },
  "devDependencies": {
    "@types/express": "^4.17.13",
    "@types/node": "^17.0.25",
    "concurrently": "^7.1.0",
    "nodemon": "^2.0.15",
    "prisma": "^3.12.0",
    "ts-node": "^10.7.0",
    "typescript": "^4.6.3"
  }
}
```

And this is my tsconfig.json:
```
{
  "compilerOptions": {
    "target": "es6",
    "lib": [
      "esnext"
    ],
    "module": "es6",
    "moduleResolution": "node",
    "sourceMap": true, 
    "outDir": "dist",
    "removeComments": true,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "skipLibCheck": true
  }
}
```

I don't think i've missed something, so can someone help me and tell me if i've done something wrong. Thanks.

### How to reproduce

Follow the official guid to the quick start of prisma 3.12.0 with Typescript and MySQL and the error will come out when you try do this:

```
import { PrismaClient } from '@prisma/client'

const prisma = new PrismaClient()
```

### Expected behavior

Do not throw an error and import correctly the PrismaClient

### Prisma information

```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

model User {
  id      Int      @id @default(autoincrement())
  email   String   @unique
  name    String?
}
```

### Environment & setup

- OS: Mac OS
- Database: MySQL
- Node.js version: 16.4.2

### Prisma Version

3.12.0

**Comment by austincrim** [comment, marked as answer]
Hey @Fede-Ferra 👋 ,

Thanks for all the details! Prisma does not currently support ESM by default, but we have an [outstanding issue](https://github.com/prisma/prisma/issues/5030) that is tracking progress and other workarounds that you might be interested in.

**Reply by ithilbor** [reply]
Thanks Austin! I've got a look at that and i found many interesting things and a little workaround for know! Maybe i'll also close this post and will do some other questions there because as i see they are very active on that discussion.
Thank you so much again!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
